src/components/controllers/repl/repl.worker.js

```diff
@@ -84,30 +84,48 @@ async function bundle(sources) {
 			{
 				name: 'repl',
 				resolveId(id, importer) {
+					// Handle local source files
 					if (isSource(id)) return id;
+
+					// Resolve absolute path ids relative to their importer. For example,
+					// esm.sh will return this internally. `https://esm.sh/preact` returns
+					// `exports * from "/stable/preact@10.18.0/es2022/preact.mjs"`
 					if (id[0] === '/') {
+						if (!importer.match(/^https?:/)) {
+							throw new Error(`Cannot resolve ${id} from ${importer}`);
+						}
+
 						try {
 							return new URL(id, importer).href;
 						} catch (e) {}
 					}
-					if (!importer || isSource(importer) || /(^\.\/|:\/\/)/.test(id)) {
+
+					// If id is already an esm.sh url, add `?external=*` to it and return
+					if (id.includes('://esm.sh/')) {
+						const url = new URL(id);
+						url.searchParams.set('external', '*');
+						return url.href;
+					}
+
+					// Leave initial import, relative imports, & other http imports alone
+					if (!importer || /(^\.\/|:\/\/)/.test(id)) {
 						return id;
 					}
-					return new URL(id, new URL(importer + '/', 'file:')).pathname.slice(
-						1
-					);
+
+					// For everything else (i.e. bare module specifiers), resolve to a
+					// package on esm.sh. We'll support any syntax & options that esm.sh
+					// supports
+					const url = new URL(id, 'https://esm.sh/');
+					url.searchParams.set('external', '*');
+					return url.href;
 				},
 				async load(id) {
 					if (isSource(id)) {
 						const code = sources[id];
 						const out = transpile(code);
 						return out;
 					}
-					if (id.startsWith('https://esm.sh/')) return get(id);
-					const [, mod, v, path] =
-						id.match(/^((?:@[^@/]+\/)?[^@/]+)(?:@(\d[^@/]*))?(?:\/(.*))?$/) ||
-						[];
-					return dep(mod, v, path);
+					return get(id);
```

Comment:
Wasn't sure what this doing but I've removed it now since we resolve all module specifiers to esm.sh URLs in `resolveId()` instead of `load()` now. As its written now, you can specify any syntax or options esm.sh supports (e.g. `import useSWR from "swr@^2?alias=react:preact/compat";`)

Reply:
Okay this particular example doesn't work cuz setting `external` breaks downstream aliasing but removing `external` doesn't work either cuz `https://esm.sh/swr@2.2.4?alias=react:preact/compat` will attempt to import `/stable/preact@10.17.1/es2022/compat.js` instead of the latest version 🤦‍♂️. Since we just published a new version, perhaps this is just some temporary version caching delay?

Reply:
Okay aliasing just doesn't work currently. If I manually specifiy 10.17.1 everywhere I can, some transitive dependency will still pull in 10.18.0 lol